newIDE/app/src/EventsFunctionsExtensionEditor/index.js

```diff
@@ -292,7 +292,36 @@ export default class EventsFunctionsExtensionEditor extends React.Component<
     );
   };
 
-  _makeRenameFreeEventsFunction = (i18n: I18nType) => (
+  _makeRenameEventsFunction = (i18n: I18nType) => (
+    eventsBasedBehavior: ?gdEventsBasedBehavior,
+    eventsBasedObject: ?gdEventsBasedObject,
+    eventsFunction: gdEventsFunction,
+    newName: string,
+    done: boolean => void
+  ) => {
+    if (eventsBasedBehavior) {
+      this._renameBehaviorEventsFunction(
+        i18n,
+        eventsBasedBehavior,
+        eventsFunction,
+        newName,
+        done
+      );
+    } else if (eventsBasedObject) {
+      this._renameObjectEventsFunction(
+        i18n,
+        eventsBasedObject,
+        eventsFunction,
+        newName,
+        done
+      );
+    } else {
+      this._renameFreeEventsFunction(i18n, eventsFunction, newName, done);
```

Comment:
I'm not familiar with this part of the code.
Should this line be called in every case?
or only when eventsBasedBehavior and eventsBasedObject are not defined like you've done?